src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -586,20 +581,41 @@ initStateInterpreter :: EvalState -> Interpreter p
 initStateInterpreter s = Interpreter $ \start end withRollback runInput ->
     withRollback (put s >> start runInput >>= end)
 
+checkTooBigTx
+    :: Gas
+    -> GasLimit
+    -> TransactionM p (CommandResult [TxLog Value])
+    -> TransactionM p (CommandResult [TxLog Value])
+checkTooBigTx initialGas gasLimit next
+  | initialGas >= (fromIntegral gasLimit) = do
+      txGasUsed .= (fromIntegral gasLimit) -- all gas is consumed
+
+      let !pe = PactError GasError def []
+            $ "Tx too big (" <> pretty initialGas <> "), limit "
+            <> pretty gasLimit
+
+      jsonErrorResult pe "Tx too big"
+  | otherwise = next
+
+gasInterpreter :: Gas -> TransactionM db (Interpreter p)
+gasInterpreter g = do
+    mc <- use txCache
+    return $ initStateInterpreter
+        $ set evalGas g
+        $ setModuleCache mc def
+
 -- | Initial gas charged for transaction size
 --   ignoring the size of a continuation proof, if present
 --
 initialGasOf :: PayloadWithText -> Gas
 initialGasOf cmd = gasFee
   where
-    feePerByte :: Decimal = 0.01
-
     contProofSize =
       case _pPayload (payloadObj cmd) of
         Continuation (ContMsg _ _ _ _ (Just (ContProof p))) -> B.length p
         _ -> 0
     txSize = SB.length (payloadBytes cmd) - contProofSize
-    gasFee = round $ fromIntegral txSize * feePerByte
+    gasFee = expLengthPenalty txSize
```

Comment:
We probably still need the factor, charging straight byte size is most likely too big.

Reply:
see latest commit